snapcraft/extractors/appstream.py

```diff
@@ -35,11 +35,25 @@ def extract(path: str) -> ExtractedMetadata:
     summary = None
     node = tree.find('summary')
     if node is not None:
-        summary = node.text
+        summary = node.text.strip()
 
     description = None
     node = tree.find('description')
     if node is not None:
-        description = node.text
-
-    return ExtractedMetadata(summary=summary, description=description)
+        description = node.text.strip()
+
+    icon = None
+    node = tree.find('icon')
+    if (node is not None and 'type' in node.attrib and
+            node.attrib['type'] == 'local'):
+        icon = node.text.strip()
```

Comment:
Can you explain this a little (perhaps with a comment)? Is it possible to have a URL here? Would it be dumb for us to download it?

Reply:
Yes, it ccould be remote instead of local.
I think it will not be dumb, but I think it must be handled somewhere else. In here we just copy the value to the icon key in config_data. So we would need to support remote URLs in config_data first. I think we should do it, as a future branch.
I will add a comment, and a bug.